src/main/java/org/fenixedu/academic/domain/degreeStructure/CourseGroup.java

```diff
@@ -842,4 +857,74 @@ public void applyToCurricularCourses(final ExecutionYear executionYear, final Pr
         }
     }
 
+    public Context findChild(final MultiLanguageString name) {
```

Comment:
Given the 'fuzzy' nature of this method, it should either be very well documented, or be given a name that better conveys what it does.